pkg/log/options.go

```diff
@@ -107,55 +124,181 @@ func DefaultOptions() *Options {
 		RotationMaxSize:    defaultRotationMaxSize,
 		RotationMaxAge:     defaultRotationMaxAge,
 		RotationMaxBackups: defaultRotationMaxBackups,
-		outputLevel:        string(defaultOutputLevel),
-		stackTraceLevel:    string(defaultStackTraceLevel),
+		outputLevels:       DefaultScopeName + ":" + levelToString[defaultOutputLevel],
+		stackTraceLevels:   DefaultScopeName + ":" + levelToString[defaultStackTraceLevel],
 		LogGrpc:            true,
 	}
 }
 
-func isValid(level Level) bool {
-	switch level {
-	case DebugLevel, InfoLevel, WarnLevel, ErrorLevel, NoneLevel:
-		return true
-	default:
-		return false
+// SetOutputLevel sets the minimum log output level for a given scope.
+func (o *Options) SetOutputLevel(scope string, level Level) {
+	sl := scope + ":" + levelToString[level]
```

Comment:
For this to work at run-time using HTTP handlers, we probably need some mutexes protecting log scope options.

Reply:
Functions on the Option type do not need to be used from an HTTP handler, those are only used once when starting a process before calling the Configure method.

What we would want to do from an HTTP handler is control the output level of individual scopes. This can be done safely by just poking fields in the Scope structure.